<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="WCB Bioinformatics Core Facility" />


<title>Bioinformatics servers and user policy</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HOME</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="training.html">Training Courses</a>
</li>
<li>
  <a href="best_practices.html">Best Practices</a>
</li>
<li>
  <a href="resources.html">Further resources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Bioinformatics servers and user policy</h1>
<h4 class="author">WCB Bioinformatics Core Facility</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#bioinformatics-servers">Bioinformatics Servers</a></li>
<li><a href="#server-access">Server access</a></li>
<li><a href="#r-and-rstudio">R and RStudio</a></li>
<li><a href="#web-server">Web server</a></li>
<li><a href="#data-policy">Data policy</a></li>
<li><a href="#data-management">Data management</a></li>
<li><a href="#server-usage">Server usage</a></li>
<li><a href="#software-and-updates">Software and updates</a></li>
<li><a href="#storage-and-backups">Storage and Backups</a></li>
<li><a href="#mailing-list">Mailing list</a></li>
<li><a href="#further-information">Further information</a></li>
</ul>
</div>

<hr />
<div id="bioinformatics-servers" class="section level2">
<h2>Bioinformatics Servers</h2>
<p>These servers are fully paid for by the Wellcome Trust Centre for
Cell Biology and are managed by the bioinformatics core facility.</p>
<p>Accounts are limited to members of WCB and centre collaborators. To
request an account please contact the bioinformatics team.</p>
<p>Once your account is active, the following servers can be accessed
over the university network or via the <a
href="https://www.ed.ac.uk/information-services/computing/desktop-personal/vpn">VPN
service</a> by logging in with your EASE credentials.</p>
<ul>
<li><p>bifx-core2.bio.ed.ac.uk (coming soon)</p>
<ul>
<li>512Gb RAM</li>
<li>32 dual CPU cores</li>
</ul></li>
<li><p>bifx-core3.bio.ed.ac.uk</p>
<ul>
<li><p>1Tb RAM</p></li>
<li><p>128 dual CPU cores</p></li>
<li><p>Command line usage (ssh)</p></li>
<li><p>R-Studio: <a href="https://bifx-rstudio.bio.ed.ac.uk/"
class="uri">https://bifx-rstudio.bio.ed.ac.uk/</a></p></li>
<li><p>R-Shiny: <a href="http://bifx-core3.bio.ed.ac.uk:3838/"
class="uri">http://bifx-core3.bio.ed.ac.uk:3838/</a></p></li>
<li><p>Apache web server</p></li>
</ul></li>
</ul>
<hr />
</div>
<div id="server-access" class="section level2">
<h2>Server access</h2>
<p>For command line access, use the ssh command from a Terminal
application in Windows, MacOS and Linux e.g.</p>
<ul>
<li><code>ssh -Y username@bifx-core3.bio.ed.ac.uk</code></li>
</ul>
<p>If you prefer to use a graphical interface the servers can be
accessed via X2Go or with the Windows Remote Desktop application.</p>
<ul>
<li><p>X2Go</p>
<ul>
<li><a
href="https://wiki.x2go.org/doku.php/doc:installation:start">Download
the X2Go client</a></li>
</ul></li>
<li><p>Remote Desktop</p>
<ul>
<li><p>Pre-installed on Windows</p></li>
<li><p><a
href="https://apps.apple.com/us/app/microsoft-remote-desktop/id1295203466?mt=12">Download
for Mac from the App store</a></p></li>
</ul></li>
</ul>
<hr />
</div>
<div id="r-and-rstudio" class="section level2">
<h2>R and RStudio</h2>
<ul>
<li><p>RStudio on bifx-core3 can be accessed at <a
href="https://bifx-rstudio.bio.ed.ac.uk"
class="uri">https://bifx-rstudio.bio.ed.ac.uk</a></p></li>
<li><p>A Shiny server is available on the university network only at <a
href="http://bifx-core3.bio.ed.ac.uk:3838"
class="uri">http://bifx-core3.bio.ed.ac.uk:3838</a></p>
<ul>
<li><p>Add your own Shiny Apps to a <code>ShinyApps</code> folder in
your home directory.</p></li>
<li><p>Launch these from <a
href="http://bifx-core3.bio.ed.ac.uk:3838/username/"
class="uri">http://bifx-core3.bio.ed.ac.uk:3838/username/</a></p></li>
</ul></li>
</ul>
<hr />
</div>
<div id="web-server" class="section level2">
<h2>Web server</h2>
<p>Data on the server can be accessed over the web (e.g. files, images,
html documents, IGV/UCSC tracks). Please contact us for details on how
to set this up.</p>
<hr />
</div>
<div id="data-policy" class="section level2">
<h2>Data policy</h2>
<ul>
<li><p>No personal data should be stored on the bifx-* servers. All data
should be related to projects associated with the Wellcome Centre unless
dispensation has been given by the centre director.</p></li>
<li><p>All data on the bifx-* servers can be viewed by members of the
bioinformatics core or the centre director.</p></li>
<li><p>As the bioinformatics core group has superuser permissions, we
can see all data on the server. Instructions on how to access the
servers are lodged with the centre director in case of emergency.
Therefore they are a de-facto administrator.</p></li>
<li><p>Your data can be accessed by your PI on request. No request is
needed if the PI has an account on the server and can view your files.
The data stored on the servers should only relate to the Wellcome Centre
PI grant holder for the projects and therefore it is reasonable that
they can access their own data.</p></li>
<li><p>Users are responsible for their own data (see backup
policy).</p></li>
<li><p>By default, when you create a new file on the server it is
created with the following permissions:</p>
<ul>
<li>Personal: read/write/execute</li>
<li>Global/group: read</li>
</ul></li>
<li><p>This means that only you can change, delete or execute the new
file, but anyone logged in to the server can view it. If you would like
to change the permissions on a file you can use the <code>chmod</code>
command. Further details on how to do this can be found by reviewing our
introduction to linux course.</p></li>
</ul>
<hr />
</div>
<div id="data-management" class="section level2">
<h2>Data management</h2>
<ul>
<li><p>Users should have a data management plan for their projects.
Please see <a
href="https://uoe.sharepoint.com/:w:/s/wcbbioinformatics/EQAVC1KoHmtBrA6rUOasQcIBisBh2fIQ3gBp7XA_3zWlxQ?e=c2xMbY">this
document</a> for advice.</p></li>
<li><p>Users should endeavour to minimise disk usage by removing
unnecessary files and compressing files where appropriate.</p></li>
<li><p>Completed projects should be archived with metadata and code to
ensure reproducibility.</p></li>
<li><p>Before you leave the university, please ensure all required data
is accessible to your group.</p></li>
<li><p>Group folders exist on the bifx servers at /groups. These can be
used for shared data and for archiving old projects/accounts.</p></li>
<li><p>Please follow our recommended <a href="best_practices.html">best
practices</a> for working on the bifx servers. This web page will be
updated regularly.</p></li>
</ul>
<hr />
</div>
<div id="server-usage" class="section level2">
<h2>Server usage</h2>
<ul>
<li><p>All users must complete our <a
href="http://bifx-core3.bio.ed.ac.uk/hyweldd/training/Bioinformatics_on_the_Command_Line/">introduction
to linux</a> course, either in their own time or by attending one of our
workshops.</p></li>
<li><p>The bioinformatics servers are a shared environment and users are
responsible for ensuring their jobs do not overwhelm computing
resources. The servers can cope with high memory usage and
multi-threading, however they will slow down when there is a high number
of read/write operations.</p>
<ul>
<li><p>Limit simultaneous jobs with large read/write requirements
(e.g. genome alignment).</p></li>
<li><p>Do not set multi-threaded jobs to use the majority of
CPUs.</p></li>
</ul></li>
<li><p>In particular, users should be aware of Unix commands to monitor
their jobs and avoid overuse of computing resources, e.g:</p>
<ul>
<li><code>ps / pgrep</code></li>
<li><code>top / htop</code></li>
<li><code>nice</code></li>
<li><code>kill</code></li>
</ul></li>
<li><p>Use the scratch space in <code>/scratch</code> to speed up
analyses/downloads/uploads. This is local disk space which will respond
quicker. However, it is <strong>NOT BACKED UP</strong> and should be
used as temporary storage only. All key datasets should be transferred
to your home space.</p></li>
</ul>
<hr />
</div>
<div id="software-and-updates" class="section level2">
<h2>Software and updates</h2>
<ul>
<li><p>The core facility will regularly maintain software and updates.
We will strive to provide adequate warning when updates may cause
interruption. All communication will be broadcast via the bifx-users
mailing list (see below).</p></li>
<li><p>Details on version control of bioinformatics software and
maintaining your own packages are available in our <a
href="best_practices.html">best practices</a> section.</p></li>
</ul>
<hr />
</div>
<div id="storage-and-backups" class="section level2">
<h2>Storage and Backups</h2>
<ul>
<li><p>Data saved in home directories is stored on the University of
Edinburgh DataStore service. Two backup systems exist:</p>
<ul>
<li>Daily ‘snapshots’ kept for two weeks to allow users to restore
data.</li>
<li>Daily tape backups kept for sixty days in case of total data
loss.</li>
</ul></li>
<li><p>Data in <code>/scratch</code> directories is not backed up. This
disk space is provided to speed up analysis but output files should be
copied back to your home directory. Please remove data from /scratch
regularly to free up space for other users. Old data may be periodically
removed by the bioinformatics core facility to free up space.</p></li>
<li><p>We recommend that users keep additional backups of key datasets
e.g:</p>
<ul>
<li><p>Personal copies on encrypted external HDD.</p>
<ul>
<li>See <a
href="https://www.ed.ac.uk/infosec/how-to-protect/encrypting">here</a>
for university advice on encrypting devices.</li>
</ul></li>
<li><p>Online research data repositories e.g.</p>
<ul>
<li><p>Sequence repositories (GEO, SRA, ArrayExpress)</p></li>
<li><p><a href="https://zenodo.org/">Zenodo</a></p></li>
</ul></li>
</ul></li>
</ul>
<hr />
</div>
<div id="mailing-list" class="section level2">
<h2>Mailing list</h2>
<p>For announcements regarding server issues and updates, please join
our mailing list. To join, send an email with no subject line to the
following address: <a href="mailto:sympa@mlist.is.ed.ac.uk"
class="email">sympa@mlist.is.ed.ac.uk</a></p>
<p>Include the following 2 lines of text (replacing FIRSTNAME and
LASTNAME with your own first name and surname) SUBSCRIBE <a
href="mailto:bifx-users@mlist.is.ed.ac.uk"
class="email">bifx-users@mlist.is.ed.ac.uk</a> FIRSTNAME LASTNAME
QUIT</p>
<hr />
</div>
<div id="further-information" class="section level2">
<h2>Further information</h2>
<p>Further information, including how to join our Slack channel and
details of our Bioinformatics Live sessions, is available on our <a
href="https://uoe.sharepoint.com/sites/wcbbifx-core">website</a>.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
