<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="WCB Bioinformatics Core Facility" />


<title>Bioinformatics servers: User policy</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 45px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h2 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h3 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h4 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h5 {
  padding-top: 50px;
  margin-top: -50px;
}
.section h6 {
  padding-top: 50px;
  margin-top: -50px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HOME</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="training.html">Training Courses</a>
</li>
<li>
  <a href="best_practices.html">Best Practices</a>
</li>
<li>
  <a href="resources.html">Further resources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Bioinformatics servers: User policy</h1>
<h4 class="author">WCB Bioinformatics Core Facility</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#servers">Servers</a></li>
<li><a href="#data-policy">Data policy</a></li>
<li><a href="#data-management">Data management</a></li>
<li><a href="#server-usage">Server usage</a></li>
<li><a href="#software-and-updates">Software and updates</a></li>
<li><a href="#storage-and-backups">Storage and Backups</a></li>
<li><a href="#mailing-list">Mailing list</a></li>
</ul>
</div>

<hr />
<div id="servers" class="section level2">
<h2>Servers</h2>
<p>These servers are fully paid for by the Wellcome Trust Centre for Cell Biology and are managed by the bioinformatics core facility.</p>
<p>Servers can be accessed via ssh and the University VPN service.</p>
<ul>
<li><p>bifx-core2.bio.ed.ac.uk (coming soon)</p></li>
<li><p>bifx-core3.bio.ed.ac.uk</p>
<ul>
<li><p>Command line usage</p></li>
<li><p>R-Studio: <a href="https://bifx-rstudio.bio.ed.ac.uk/" class="uri">https://bifx-rstudio.bio.ed.ac.uk/</a></p></li>
<li><p>R-Shiny: <a href="http://bifx-core3.bio.ed.ac.uk:3838/" class="uri">http://bifx-core3.bio.ed.ac.uk:3838/</a></p></li>
<li><p>Apache web server</p></li>
</ul></li>
</ul>
</div>
<div id="data-policy" class="section level2">
<h2>Data policy</h2>
<ul>
<li><p>No personal data should be stored on the bifx-* servers. All data should be related to projects associated with the Wellcome Centre unless dispensation has been given by the centre director.</p></li>
<li><p>All data on the bifx-* servers can be viewed by members of the bioinformatics core or the centre director.</p></li>
<li><p>As the bioinformatics core group has superuser permissions, we can see all data on the server. Instructions on how to access the servers are lodged with the centre director in case of emergency. Therefore they are a de-facto administrator.</p></li>
<li><p>Your data can be accessed by your PI on request. No request is needed if the PI has an account on the server and can view your files. The data stored on the servers should only relate to the Wellcome Centre PI grant holder for the projects and therefore it is reasonable that they can access their own data.</p></li>
<li><p>Users are responsible for their own data (see backup policy).</p></li>
<li><p>Default umask is 022, you should be aware what this means and how to change this if desired:</p>
<ul>
<li>Personal: read/write/execute</li>
<li>Global/group: read</li>
</ul></li>
</ul>
</div>
<div id="data-management" class="section level2">
<h2>Data management</h2>
<ul>
<li><p>Users should have a data management plan for their projects. Please see <a href="https://uoe.sharepoint.com/:w:/s/wcbbioinformatics/EQAVC1KoHmtBrA6rUOasQcIBisBh2fIQ3gBp7XA_3zWlxQ?e=c2xMbY">this document</a> for advice.</p></li>
<li><p>Users should endeavour to minimise disk usage by removing unnecessary files and compressing files where appropriate.</p></li>
<li><p>Completed projects should be archived with metadata and code to ensure reproducibility.</p></li>
<li><p>Before you leave the university, please ensure all required data is accessible to your group.</p></li>
<li><p>Group folders exist on the bifx servers at /groups. These can be used for shared data and for archiving old projects/accounts.</p></li>
<li><p>Please follow our recommended <a href="best_practices.html">best practices</a> for working on the bifx servers. This web page will be updated semi-regularly.</p></li>
</ul>
</div>
<div id="server-usage" class="section level2">
<h2>Server usage</h2>
<ul>
<li><p>All users must complete our <a href="http://bifx-core3.bio.ed.ac.uk/hyweldd/training/Bioinformatics_on_the_Command_Line/">introduction to linux</a> course, either in their own time or by attending one of our workshops.</p></li>
<li><p>The bioinformatics servers are a shared environment and users are responsible for ensuring their jobs do not overwhelm computing resources. The servers can cope with high memory usage and multi-threading, however they will slow down when there is a high number of read/write operations.</p>
<ul>
<li><p>Limit simultaneous jobs with large read/write requirements (e.g. genome alignment).</p></li>
<li><p>Do not set multi-threaded jobs to use the majority of CPUs.</p></li>
</ul></li>
<li><p>In particular, users should be aware of Unix commands to monitor their jobs and avoid overuse of computing resources, e.g:</p>
<ul>
<li><code>ps / pgrep</code></li>
<li><code>top / htop</code></li>
<li><code>nice</code></li>
<li><code>kill</code></li>
</ul></li>
<li><p>Use the scratch space in <code>/scratch</code> to speed up analyses/downloads/uploads. This is local disk space which will respond quicker. However, it is <strong>NOT BACKED UP</strong> and should be used as temporary storage only. All key datasets should be transferred to your home space.</p></li>
</ul>
</div>
<div id="software-and-updates" class="section level2">
<h2>Software and updates</h2>
<ul>
<li><p>The core facility will regularly maintain software and updates. We will strive to provide adequate warning when updates may cause interruption. All communication will be broadcast via the bifx-users mailing list (see below).</p></li>
<li><p>Details on version control of bioinformatics software and maintaining your own packages are available in our <a href="best_practices.html">best practices</a> section.</p></li>
</ul>
</div>
<div id="storage-and-backups" class="section level2">
<h2>Storage and Backups</h2>
<ul>
<li><p>Data saved in home directories is stored on the University of Edinburgh DataStore service. Two backup systems exist:</p>
<ul>
<li><p>Daily ‘snapshots’ kept for two weeks to allow users to restore data.</p></li>
<li><p>Daily tape backups kept for sixty days in case of total data loss.</p></li>
</ul></li>
<li><p>Data in <code>/scratch</code> directories is not backed up. This disk space is provided to speed up analysis but output files should be copied back to your home directory. Please remove data from /scratch regularly to free up space for other users.</p></li>
<li><p>We recommend that users keep additional backups of key datasets e.g:</p>
<ul>
<li>Personal copies on external HDD.</li>
<li>Online data repositories</li>
</ul></li>
</ul>
</div>
<div id="mailing-list" class="section level2">
<h2>Mailing list</h2>
<p>To join our mailing list, please send an email with no subject line to the following address: <a href="mailto:sympa@mlist.is.ed.ac.uk" class="email">sympa@mlist.is.ed.ac.uk</a></p>
<p>Include the following 2 lines of text (replacing FIRSTNAME and LASTNAME with your own first name and surname) SUBSCRIBE <a href="mailto:bifx-users@mlist.is.ed.ac.uk" class="email">bifx-users@mlist.is.ed.ac.uk</a> FIRSTNAME LASTNAME QUIT</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
